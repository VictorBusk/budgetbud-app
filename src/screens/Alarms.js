import React, {PureComponent} from 'react';
import {FlatList} from 'react-native';
import {connect} from 'react-redux';
import {
    AppHeader,
    Separator
} from "../components/";
import {
    Container,
    ListItem,
    Body,
    Right,
    Text,
    Spinner,
    CheckBox,
    Button
} from 'native-base';
import I18n from "../strings/i18n";
import {
    getAccounts,
    linkAccounts,
    accountsSelected
} from "../actions";
import {
    button,
    text,
    list,
    container, color
} from "../style";

class Alarms extends PureComponent {
    render() {
        return (
            <Container style={container.signedInContainer}>
                <AppHeader headerText={I18n.t('alarmsHeader')}
                           showBackButton={true}
                           onLeftButtonPress={() => this.props.navigation.pop()}/>

                <Container style={{flex: 4, justifyContent: 'center'}}>
                </Container>

                <Separator/>

                <Button rounded
                        onPress={() => {
                            if (!this.props.enableLoading) {
                                this.onSavePress()
                            }
                        }}
                        style={[button.defaultButton, color.button]}
                >
                    {this.props.enableLoading ? (
                        <Spinner color='#D0D0D0'/>) : (
                        <Text style={text.submitButtonText}>{I18n.t('alarmsSaveButton')}</Text>
                    )}
                </Button>
            </Container>
        );
    }

    renderItem = ({item}) => {
        const checked = '';

        return (
            <ListItem>
                <Body>
                <Text style={color.text}>{item.name}</Text>
                </Body>
                <Right style={list.listItemCheckBoxPadding}>
                    <CheckBox
                        style={checked ? color.checkboxChecked : color.checkboxUnchecked}
                        checked={checked}
                        onPress={() => this.onCheckBoxPress(item)}
                    />
                </Right>
            </ListItem>
        );
    }
}

const mapStateToProps = ({alarm}) => {
    const {enableLoading} = alarm;

    return {
        enableLoading
    };
};

const mapDispatchToProps = {
};

export default connect(mapStateToProps, mapDispatchToProps)(Alarms);
