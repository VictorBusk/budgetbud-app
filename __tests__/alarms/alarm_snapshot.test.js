import React from 'react';
import configureStore from 'redux-mock-store';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Alarms from "../../src/app/alarms/Alarms";
import {INITIAL_ALARM_STATE, INITIAL_BUDGET_STATE} from "../test_helper/initial_state";

Enzyme.configure({adapter: new Adapter()});

const middlewares = [];
const mockStore = configureStore(middlewares);

describe('Alarms', () => {
    it('renders initially as expected', () => {
        const wrapper = shallow(
            <Alarms/>,
            {context: {store: mockStore({
                alarm: INITIAL_ALARM_STATE,
                budget: INITIAL_BUDGET_STATE
            })}},
        );
        expect(wrapper.dive()).toMatchSnapshot();
    });

    it('renders budgetExceeded = true as expected', () => {
        let newState = INITIAL_ALARM_STATE;
        newState.budgetExceeded = true;
        const wrapper = shallow(
            <Alarms/>,
            {context: {store: mockStore({
                alarm: INITIAL_ALARM_STATE,
                budget: INITIAL_BUDGET_STATE
            })}},
            );
        expect(wrapper.dive()).toMatchSnapshot();
    });
});
