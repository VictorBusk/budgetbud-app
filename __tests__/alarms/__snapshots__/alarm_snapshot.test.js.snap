// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alarms renders budgetExceeded = true as expected 1`] = `
<Styled(Container)
  style={
    Object {
      "backgroundColor": "#ffffff",
    }
  }
>
  <AppHeader
    headerText="Alarms"
    infoText="Here you can choose which alarms you want enabled. If an alarm is activated, you will receive a notification. Press [Save] to save the changes."
    onLeftButtonPress={[Function]}
    showBackButton={true}
  />
  <Styled(Container)
    style={
      Object {
        "justifyContent": "center",
      }
    }
  >
    <Styled(Content)>
      <Styled(List)>
        <Styled(ListItem)>
          <Styled(Body)>
            <Styled(Text)
              style={
                Object {
                  "color": "#003755",
                }
              }
            >
              Notify if budget is exceeded
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)
            style={
              Object {
                "paddingRight": 15,
              }
            }
          >
            <Styled(CheckBox)
              checked={true}
              onPress={[Function]}
              style={
                Object {
                  "backgroundColor": "#295E80",
                  "borderColor": "#003755",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Body)>
            <Styled(Text)
              style={
                Object {
                  "color": "#003755",
                }
              }
            >
              Notify weekly status
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)
            style={
              Object {
                "paddingRight": 15,
              }
            }
          >
            <Styled(CheckBox)
              checked={false}
              onPress={[Function]}
              style={
                Object {
                  "backgroundColor": "#F6F7F9",
                  "borderColor": "#003755",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
      </Styled(List)>
    </Styled(Content)>
  </Styled(Container)>
  <Separator />
  <Styled(Button)
    onPress={[Function]}
    rounded={true}
    style={
      Array [
        Object {
          "alignSelf": "center",
          "height": 40,
          "justifyContent": "center",
          "marginBottom": 20,
          "marginTop": 20,
          "width": "90%",
        },
        Object {
          "backgroundColor": "#00263A",
        },
      ]
    }
  >
    <Styled(Text)
      style={
        Object {
          "alignSelf": "center",
          "color": "white",
          "fontWeight": "600",
        }
      }
    >
      Save
    </Styled(Text)>
  </Styled(Button)>
</Styled(Container)>
`;

exports[`Alarms renders initially as expected 1`] = `
<Styled(Container)
  style={
    Object {
      "backgroundColor": "#ffffff",
    }
  }
>
  <AppHeader
    headerText="Alarms"
    infoText="Here you can choose which alarms you want enabled. If an alarm is activated, you will receive a notification. Press [Save] to save the changes."
    onLeftButtonPress={[Function]}
    showBackButton={true}
  />
  <Styled(Container)
    style={
      Object {
        "justifyContent": "center",
      }
    }
  >
    <Styled(Content)>
      <Styled(List)>
        <Styled(ListItem)>
          <Styled(Body)>
            <Styled(Text)
              style={
                Object {
                  "color": "#003755",
                }
              }
            >
              Notify if budget is exceeded
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)
            style={
              Object {
                "paddingRight": 15,
              }
            }
          >
            <Styled(CheckBox)
              checked={false}
              onPress={[Function]}
              style={
                Object {
                  "backgroundColor": "#F6F7F9",
                  "borderColor": "#003755",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Body)>
            <Styled(Text)
              style={
                Object {
                  "color": "#003755",
                }
              }
            >
              Notify weekly status
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)
            style={
              Object {
                "paddingRight": 15,
              }
            }
          >
            <Styled(CheckBox)
              checked={false}
              onPress={[Function]}
              style={
                Object {
                  "backgroundColor": "#F6F7F9",
                  "borderColor": "#003755",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
      </Styled(List)>
    </Styled(Content)>
  </Styled(Container)>
  <Separator />
  <Styled(Button)
    onPress={[Function]}
    rounded={true}
    style={
      Array [
        Object {
          "alignSelf": "center",
          "height": 40,
          "justifyContent": "center",
          "marginBottom": 20,
          "marginTop": 20,
          "width": "90%",
        },
        Object {
          "backgroundColor": "#00263A",
        },
      ]
    }
  >
    <Styled(Text)
      style={
        Object {
          "alignSelf": "center",
          "color": "white",
          "fontWeight": "600",
        }
      }
    >
      Save
    </Styled(Text)>
  </Styled(Button)>
</Styled(Container)>
`;
